x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/create_policy_extension.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { memo } from 'react';
+import { EuiForm } from '@elastic/eui';
+import type { PackagePolicyCreateExtensionComponentProps } from '@kbn/fleet-plugin/public';
+import { CLOUDBEAT_EKS } from '../../../common/constants';
+import { DeploymentTypeSelect, InputType } from './deployment_type_select';
+import { EksFormWrapper } from './eks_form';
+import { getEnabledInputType, getUpdatedDeploymentType, getUpdatedEksVar } from './utils';
+
+export const CspCreatePolicyExtension = memo<PackagePolicyCreateExtensionComponentProps>(
```

Comment:
this and the edit form are the same except the edit one has the deployment selector disabled. 
their component/test are mostly duplicated and we should address that 
but i didn't want to pass `disableDeploymentTypeSelector` to a custom form so for a first iteration this is good enough IMO

Reply:
for reference: each is a different context - `edit` / `create` and instead of combining a single view for both i think it's preferable to make each of those views composed from their shared and non shared components.